src/UglyToad.PdfPig/Graphics/Colors/ColorSpaceDetails.cs

```diff
@@ -954,15 +983,17 @@ private RGBColor TransformToRGB((double A, double B, double C) colorAbc)
         }
 
         /// <inheritdoc/>
-        internal override IReadOnlyList<byte> Transform(IReadOnlyList<byte> decoded)
+        internal override ReadOnlySpan<byte> Transform(ReadOnlySpan<byte> decoded)
         {
-            var transformed = new List<byte>();
-            for (var i = 0; i < decoded.Count; i += 3)
+            var transformed = new byte[decoded.Length];

```

Comment:
Note to self - Check if that always work

Reply:
should have the identical behavior as the List, without any additional copying as the list grows.